app/server/models/custom_attribute_user_storage.py

```diff
@@ -1,26 +1,36 @@
 from server import db
 from server.models.utils import ModelBase
+from server.models.custom_attribute import CustomAttribute
 from sqlalchemy import cast
+from sqlalchemy.ext.hybrid import hybrid_property
 
 class CustomAttributeUserStorage(ModelBase):
     __tablename__ = 'custom_attribute_user_storage'
 
-    name = db.Column(db.String)
-    value = db.Column(db.JSON)
+    custom_attribute_id = db.Column(db.Integer, db.ForeignKey('custom_attribute.id'))
+    custom_attribute = db.relationship("CustomAttribute", back_populates="attributes")
+
+    name = db.Column(db.String) # Deprecated for key
+    value = db.Column(db.String)
 
     user_id = db.Column(db.Integer, db.ForeignKey('user.id'))
     uploaded_image_id = db.Column(db.Integer)
 
+    @hybrid_property
+    def key(self):
+        return self.custom_attribute.name
+
     @staticmethod
     def get_attributes_and_options():
         # Get all custom attributes
-        attributes = [u.name for u in db.session.query(CustomAttributeUserStorage.name).distinct()]
-        # Get all possible options for those attributes
+        attributes = db.session.query(CustomAttribute).all()
         attribute_options = {}
+        # Get all possible options for those attributes
         for a in attributes:
-            attribute_options[a] = [
-                u[0].replace('"', '') for u in db.session.query(cast(CustomAttributeUserStorage.value, db.String))
-                .filter(CustomAttributeUserStorage.name == a)
-                .distinct()
-            ]
+            options = db.session.query(CustomAttributeUserStorage.value)\
+                .filter(CustomAttributeUserStorage.custom_attribute_id == a.id)\
```

Comment:
is this crazy expensive since it has to go over every custom attribute for every user, or is distinct somewhat kind?